modules/oe_corporate_countries_address/tests/src/Kernel/AvailableCountriesSubscriberTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Drupal\Tests\oe_corporate_countries_address\Kernel;
+
+use Drupal\entity_test\Entity\EntityTest;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\Tests\oe_corporate_countries\Kernel\CorporateCountriesRdfKernelTestBase;
+
+/**
+ * Tests the AvailableCountriesSubscriber class.
+ */
+class AvailableCountriesSubscriberTest extends CorporateCountriesRdfKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'entity_test',
+    'address',
+    'oe_corporate_countries_address',
+  ];
+
+  /**
+   * Tests the deprecated countries are removed from the countries array.
+   */
+  public function testGetAvailableCountriesArray(): void {
+
+    $entityTypeName = 'entity_test';
+
+    FieldStorageConfig::create([
+      'field_name' => 'field_country',
+      'entity_type' => $entityTypeName,
+      'type' => 'address_country',
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'My field',
+      'field_name' => 'field_country',
+      'entity_type' => $entityTypeName,
+      'bundle' => $entityTypeName,
+    ])->save();
+
+    // Get the actual countries from field.
+    // This doesn't have the deprecated countries.
+    $entity = EntityTest::create();
+    $actual_countries = $entity->get('field_country')
+      ->appendItem()
+      ->getAvailableCountries();
+
+    $expected_countries = [
+      "XK" => "XK",
+      "AX" => "AX",
+      "BE" => "BE",
+      "TF" => "TF",
+      "IT" => "IT",
+    ];
+
+    // Assert that the returned array adheres to the correct format.
+    // Assert that the array is correctly keyed.
+    // Assert that the array doesn't include deprecated countries, notably 'AN'.
```

Comment:
I don't see this assertion anywhere.

Reply:
Here my comments about this:

The line 

```
$this->assertEquals($expected_countries, $actual_countries);
``` 

performs a simple equality check. 

It asserts that `$expected_countries` and `$actual_countries` are equal. Here is what it checks:

1. Array Format:   
It checks that both arrays have the same keys and that those keys have the same values. 
So indirectly this line does indeed assert that the returned `$actual_countries` array adheres to the correct format present in `$expected_countries`.

2. Correctly Keyed:  
It checks that both arrays have exactly the same keys. The keys in `$actual_countries` must exactly match the keys in `$expected_countries`.

3. Deprecated Countries:
assertEquals itself does not inherently know about them but it simply compares two given arrays for equality. 
Because `$expected_countries` does not contain deprecated countries like `'AN'` he equality assertion would fail when `$actual_countries` has a deprecated country.

Reply:
I have removed the $this->assertEquals($expected_countries, $actual_countries); and converted it into 3 independent assertions.